### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by ericmania** [jasp-stats/jasp-issues#2306]
### JASP Version

0.18.0

### JASP Module

Descriptives

### What analysis are you seeing the problem on?

Mode

### What OS are you seeing the problem on?

Windows 10, Windows 11

### Bug Description

[MCT and Spread Exlample Data (1).csv](https://github.com/jasp-stats/jasp-issues/files/12730170/MCT.and.Spread.Exlample.Data.1.csv)

In the attached data file I had students compute mean, median, and mode for the variable "ScoreForMode". The mode returned appears incorrect. The mode should be 8.  Instead the mode returned by JASP is 7.854 with a note saying "more than one mode exists. For nominal and ordinal data the first mode is reported. For continuous data, the mode with the highest density estimate is reported, but multiple modes may exist. We recommend visualizing the data to check for multimodality."

The screen shot attached below shows the resulting output and message.

![Screenshot 2023-09-26 113621](https://github.com/jasp-stats/jasp-issues/assets/146119793/4f5f69ef-3a64-4a44-9dbb-62330f244573)

Analyses of other variables in the data set also returned modes that were incorrect. 

Multiple students using different computers running the latest version of JASP experienced the same problem.

### Expected Behaviour

The mode should come out to be 8. Here is a screenshot of the same analysis run with the same datafile using JASP version 16.3 on Microsoft 10. It shows the mode to be 8 with no message underneath the descriptive statistics table. 

![Screenshot 2023-09-26 113621](https://github.com/jasp-stats/jasp-issues/assets/146119793/5dcdb12f-211d-415f-9fdb-8b09fa1feb75)

### Steps to Reproduce

1.Open the data file MCT and Spread Example Data (1).csv
2.Go to descriptives and request mode, median, and mean for ScoreForMode
3.Observe a mode of 7.584 and compare to the scores on this variable which clearly have a mode of 8. 
...

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if possible.
- [X] I have included a JASP file (zipped) or data file that causes the crash/bug, if applicable.
- [X] I have accurately described the bug, and steps to reproduce it.

**Comment by ericmania** [comment]
![ScreenShot with correct Mode from 16 3](https://github.com/jasp-stats/jasp-issues/assets/146119793/c4bbb3e7-9d5a-4dc1-9f01-2b921a2f426d)
The wrong screen shot appears under the Expected Behaviour section above. This is the correct screenshot showing that JASP 16.3 correctly returns a mode of 8.

**Comment by paulamarentette** [comment]
Following up on Eric's comment. I am experiencing this in class too. I have this in both .18 and .18.1. 
I don't know how to get the "correct" screenshot that Eric posted. I am also uncertain why it is saying that more than one mode exists is this extremely skewed and unimodal dataset.
Using a Mac running Sonoma 14.0. But I get this on all student laptops and the PC desktop in the classroom. It is a robust error. This example seems like a rounding error, but I get this mode problem regardless of the data entered.

[Archive.zip](https://github.com/jasp-stats/jasp-issues/files/12832703/Archive.zip)
![ModeError](https://github.com/jasp-stats/jasp-issues/assets/48217084/7ed123f7-e3e8-4a12-9918-f735db1c1c8e)

**Comment by vandenman** [comment]
The mode is currently computed as if variables are scale. So for ordinal and nominal variables this is incorrect. This is a bug and will be fixed in the next release, where the calculation of the mode will depend on the type of the data (scale or ordinal/nominal).

> I am also uncertain why it is saying that more than one mode exists is any extremely skewed and unimodal dataset.

Consider the following example:
![image](https://github.com/jasp-stats/jasp-issues/assets/21319932/babf868c-ad94-42d2-a61e-7c170e204ac9)

The data is generated from a standard normal distribution, but the `mode` still indicates there may be multiple modes. I highlighted in green what the possible modes are. Basically, we use the default density estimator and test if there are multiple peaks.

**Comment by EJWagenmakers** [comment]
But the highest one is the mode, isn't it? In other words, I think we should just give the highest density value as the mode.

**Comment by paulamarentette** [comment]
Yes, I'm with the idea that the highest one is the mode. There is only one
most frequent score in that set. In my mind multiple modes exist if there
are multiple values that have the same frequency. Otherwise mode means
something different than the most frequent score(s). Can we report a
density stat rather than using the word MODE for that idea with scale
variables.

The dataset I submitted clearly only has one mode, but still makes this
claim. It is a scale variable, but by anyone's definition, unimodal.
Paula

**Comment by vandenman** [comment]
> But the highest one is the mode, isn't it? In other words, I think we should just give the highest density value as the mode.

@EJWagenmakers If there might be multiple modes, we report the one with the highest density and show the footnote, so we already do that.


@paulamarentette , it looks like the default density estimator goes astray for your data example:

![image](https://github.com/jasp-stats/jasp-issues/assets/21319932/841e4f60-234f-4ace-9b7f-071b75d77e52)

which is a different way the same problem can manifest itself I guess.

> Otherwise mode means something different than the most frequent score(s). 

That definition only applies to discrete random variables. From [Wikipedia](https://en.wikipedia.org/wiki/Mode_(statistics))

> A mode of a continuous probability distribution is often considered to be any value x at which its probability density function has a locally maximum value, so any peak is a mode.


Somehow I don't expect that we can find a default behavior that makes everybody happy. The new behavior came about because someone pointed out that our mode is incorrect for continuous variables!

@EJWagenmakers how about we add a checkbox underneath the one for mode that says, `Treat scale variables as ordinal` underneath `Mode`? That would then allow @paulamarentette to compute the mode in the way they want, without forcing them to change the type of the variable to ordinal or nominal (which would not allow them to use it as a covariate in e.g., regression).

**Comment by paulamarentette** [comment]
Thank you, this conversation is helpful to me. As a first year stats
instructor I don't cover density or local maxima and frankly never
considered that these would also be considered as modes. I very much
appreciate the responses.
My fellow stats instructors and I love JASP. Thank you for all the work you
do to put together this great software.

Best,
Paula
